packages/mm-backend/src/common/utils/memo-decoders.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  ArbitrageMemoAction,
+  MarketMakingMemoAction,
+  SpotExchange,
+  SpotOrderType,
+  TradingType,
+} from '../enums/memo.enum';
+import {
+  ArbitrageMemoDetails,
+  MarketMakingMemoDetails,
+  SpotMemoDetails,
+} from '../interfaces/memo-details.interface';
+import { PairsEnum } from '../enums/pairs.enum';
+
+export const decodeSpotMemo = (decodedMemo: string): SpotMemoDetails => {
+  if (!decodedMemo) {
+    return null;
+  }
+  const parts = decodedMemo.split(':');
+  const [
+    tradingType,
+    spotOrderType,
+    exchange,
+    destId,
+    limitPriceOrRefId,
+    refId,
+  ] = parts;
+  return {
+    tradingType: TradingType[tradingType],
+    spotOrderType: SpotOrderType[spotOrderType],
+    exchangeName: SpotExchange[exchange],
+    destId: destId as PairsEnum,
+    limitPrice: limitPriceOrRefId,
+    refId: refId,
+  };
+};
+
+export const decodeArbitrageMemo = (
+  decodedMemo: string,
```

Comment:
What's that?

Reply:
It's excluded to another branch: Fork-issue_17-snapshot-processing